task/idle/src/main.rs

```diff
@@ -33,8 +30,8 @@ fn main() -> ! {
         }
 
         // RISC-V has wfi, but unfortunately it is an illegal instruction if
-        // called from User mode, so instead we spin on a timer call.
-        #[cfg(any(target_arch = "riscv64", target_arch = "riscv32"))]
-        while sys_get_timer().now > 0 {}
+        // called from User mode.
+        #[cfg(any(target_arch = "riscv32", target_arch = "riscv64"))]
+        unsafe { riscv::asm::nop(); } // Heed the warning above.
```

Comment:
This change is not related to the series. Ideally, we should push unrelated changes as separate PRs so as not to silently change behavior under a different title. Anyways, I think this is fine for now here, but just a comment for the future.